action.yml

```diff
@@ -52,117 +55,22 @@ runs:
-        environment: ${{ inputs.environment }}
-        image_uri: ${{ steps.get-image-uri.outputs.image-uri }}
-        deployment_tag: ${{ inputs.deployment_tag }}
-        aws_access_key_id: ${{ inputs.aws_access_key_id }}
-        aws_secret_access_key: ${{ inputs.aws_secret_access_key }}
-        aws_region: ${{ inputs.aws_region }}
-
-    # Service is managed by Pulumi, only desired count and task definition should be updated here
-    - name: Update service
-      shell: bash
-      run: |
-        aws ecs update-service \
-          --task-definition "${{ steps.deploy-production-task-definition.outputs.latest-task-definition-arn }}" \
-          --desired-count ${{ inputs.desired_count }} \
-          --cluster ${{ inputs.environment }} \
-          --service ${{ inputs.ecr_repository }}-${{ inputs.environment }}
-
-    - name: Wait for service stability
+    - name: AWS ECS Task Deploy
       shell: bash
-      id: check-service-stability
+      id: run-ecs-deploy
       working-directory: ${{ github.action_path }}
       run: |
-        poetry run actions_helper wait-for-service-stability \
-          --cluster ${{ inputs.environment }} \
-          --service ${{ inputs.ecr_repository }}-${{ inputs.environment }}
-
-    - name: Determine which pair of task definitions should be deregistered
-      if: always()
-      shell: bash
-      id: determine-task-definitions-to-deregister
-      env:
-        PREVIOUS_PRODUCTION_TASK_DEFINITION: ${{ steps.deploy-production-task-definition.outputs.previous-task-definition-arn }}
-        LATEST_PRODUCTION_TASK_DEFINITION: ${{ steps.deploy-production-task-definition.outputs.latest-task-definition-arn }}
-      run: |
-        PRIMARY_DEPLOYMENT_DEFINITION_ARN=$(
-          aws ecs describe-services \
-            --cluster ${{ inputs.environment }} \
-            --services ${{ inputs.ecr_repository }}-${{ inputs.environment }} \
-            | jq -r '.services[].deployments[] | select(.status == "PRIMARY") | .taskDefinition'
-        )
-        
-        if [[ "$PRIMARY_DEPLOYMENT_DEFINITION_ARN" == "$LATEST_PRODUCTION_TASK_DEFINITION" ]]; then  
-              # Do not deregister task definition for initial deployment
-              if [ ! -z "$PREVIOUS_PRODUCTION_TASK_DEFINITION" ]; then   
-                echo "TASK_DEFINITION_PRODUCTION=$PREVIOUS_PRODUCTION_TASK_DEFINITION" >> $GITHUB_ENV
-                echo "TASK_DEFINITION_LOCAL=${{ steps.deploy-local-task-definition.outputs.previous-task-definition-arn }}" >> $GITHUB_ENV
-                echo "TASK_DEFINITION_PREFLIGHT=${{ steps.deploy-preflight-task-definition-and-run-task.outputs.previous-task-definition-arn }}" >> $GITHUB_ENV
-                echo "fail-pipeline=false" >> $GITHUB_OUTPUT
-              fi
+        if [[ "${{ inputs.allow_feature_branch_deployment }}" == "true" ]]; then
+          imageTag=$(echo ${{ github.ref_name }} | awk '{print tolower($0)}' | sed -e 's|/|-|g')
```

Comment:
Why not passing the argument to the Python script and do everything in Python?

Reply:
We discussed this, and decided to keep this part as is. As a safeguard, within the `ecs-deploy` command, we have agreed to allow feature branch deployments only for our dev environment.